isis/src/control/objs/BundleAdjust/BundleAdjust.cpp

```diff
@@ -2365,6 +2373,225 @@ namespace Isis {
+      
+      int numMeasures = point->numberOfMeasures();
+      for (int j = 0; j < numMeasures; j++) {
+        BundleMeasureQsp measure = point->at(j);
+
+        // compute ai*xi
+        static LinearAlgebra::Vector aixi(2);
+        
+        // compute qvv=(1-ak*xi)/pi
+        static LinearAlgebra::Vector qvv(2);
+
+        // calculate test statistics
+        double vx = measure->sampleResidual();
+        double vy = measure->lineResidual();
+        //double wx = fabs(vx)/(sigma0*sqrt(qvv[0]));
+        //double wy = fabs(vy)/(sigma0*sqrt(qvv[1]));
+        // attach wi to BundleMeasure -> grossOutlierTestStatistic
+        //measure.setGrossOutlierTestStatistic(wx,wy);
+      }
+    }
+
+    return true;
+  }
+
+
+  /**
+   * Flags outlier measures and control points by comparing
+   * the observations normalized residual to a critical
+   * value determined by t-distribution and a user-defined
+   * probability level.
+   *
+   * @return @b bool If the flagging was successful.
+   *
+   * @TODO How should we handle points with few measures.
+   */
+  bool BundleAdjust::flagGrossOutliers() {
+    int numRejected;
+    int totalNumRejected = 0;
+
+    int maxResidualIndex;
+    double maxResidual;
+
+    double wx, wy;
+    double usedRejectionLimit = m_bundleResults.grossRejectionLimit();
+
+
+    // TODO What to do if usedRejectionLimit is too low?
+
+    int numComingBack = 0;
+
+    int numObjectPoints = m_bundleControlPoints.size();
+
+    outputBundleStatus("\n");
+    for (int i = 0; i < numObjectPoints; i++) {
+      BundleControlPointQsp point = m_bundleControlPoints.at(i);
+
+      point->zeroNumberOfRejectedMeasures();
+
+      numRejected = 0;
+      maxResidualIndex = -1;
+      maxResidual = -1.0;
+
+      int numMeasures = point->numberOfMeasures();
+      for (int j = 0; j < numMeasures; j++) {
+
+        BundleMeasureQsp measure = point->at(j);
+
+        // grab test statistics
+        //wx = measure.grossOutlierLineTestStatistic();
+        //wy = measure.grossOutlierSampleTestStatistic();
+        wx = 1.34;
+        wy = 2.58;
+
+        // measure is good
+        if ( wx <= usedRejectionLimit && 
+             wy <= usedRejectionLimit) {
+
+          // was it previously rejected?
+          if ( measure->isRejected() ) {
+            QString status = "Coming back in: ";
+            status.append(QString("%1").arg(point->id().toLatin1().data()));
+            status.append("\r");
+            outputBundleStatus(status);
+            numComingBack++;
+            m_controlNet->DecrementNumberOfRejectedMeasuresInImage(measure->cubeSerialNumber());
+          }
+
+          measure->setRejected(false);
+          continue;
+        }
+
+        // if it's still rejected, skip it
+        if ( measure->isRejected() ) {
+          numRejected++;
+          totalNumRejected++;
+          continue;
+        }
+
+        if ( wx > maxResidual ) {
+          maxResidual = wx;
```

Comment:
Don't call this maxResidual. Call this maxTestStat or something more appropriate.